### PR TITLE
Added article intro image as first priority for og image

### PR DESCRIPTION
**Pull request by yvesh**

### CODE REVIEW

File: joomlarrssb.php [hunk 1]
```diff
@@ -184,15 +184,27 @@ public function onContentAfterTitle($context, &$article, &$params, $page)
...
+			}
 
-		// Always run this preg_match as the results are also used in the layout
-		$pattern = "/<img[^>]*src\=['\"]?(([^>]*)(jpg|gif|JPG|png|jpeg))['\"]?/";
-		preg_match($pattern, $article->text, $matches);
+			// Always run this preg_match as the results are also used in the layout
+			$pattern = "/<img[^>]*src\=['\"]?(([^>]*)(jpg|gif|JPG|png|jpeg))['\"]?/";
+			preg_match($pattern, $article->text, $matches);
+
+			$imageOg = $matches[1];
```
Comment (mbabker) [inline joomlarrssb.php]: This should check if `$matches[1]` is defined also as it wouldn't be if the article text doesn't contain any matches.

File: joomlarrssb.php [hunk 2]
```diff
@@ -184,15 +184,27 @@ public function onContentAfterTitle($context, &$article, &$params, $page)
 		$siteURL = substr(JUri::root(), 0, -1);
 		$itemURL = $siteURL . JRoute::_(ContentHelperRoute::getArticleRoute($article->slug, $article->catid));
 
-		// Get the content and merge in the template; first see if $article->text is defined
-		if (!isset($article->text))
+		// Check if we have an intro text image (First priority)
+		$article->images = json_decode($article->images);
```
Comment (mbabker) [inline joomlarrssb.php]: Should probably check and make sure `$article->images` is actually defined first.  Also, I wouldn't suggest overwriting the article's property with the decoded object, so just make it a local variable.

**Comment by mbabker** [comment]
There's one other gotcha in this, otherwise at a first glance this looks pretty good.

[In the layout](https://github.com/joomla-projects/plg_content_joomlarrssb/blob/master/tmpl/default.php) the regex matches array is also used for Pinterest sharing (that's what the `Always run this preg_match as the results are also used in the layout` comment is referring to), so the layout will need to be changed to use your new `$imageOg` variable instead of the `$matches` array since that won't always be defined now.

**Comment by yvesh** [comment]
Fixed, sorry should read comments and not do 30 things at the same time..

**Comment by mbabker** [comment]
It's all good.  That's what review and inline comments are for (mostly so I remember what features are there 😆 )

**Pull request merged**